opensearch-project/OpenSearch-Dashboards
# Fix filter actions on data table vis cells

### Description
This PR replaces the rows used in the Data Table visualization when building the cell's filter for/out buttons with the sorted ones. 

### Issues Resolved
Fixes #4836

## Screenshot
![osd-table-filters-fixed](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/26812577/bd06309c-bf05-4a05-96c9-62a95e7a3031)

## Testing the changes
- Create a data table visualization
- Change the sorting order on one of the columns
- Click on the filter for value button inside a cell
- The filter should match the text on the clicked cell

### Check List
